src/bin/genemichaels.rs

```diff
@@ -199,7 +237,104 @@ fn main() {
             Offable::On(_) => true,
         },
     };
-    if args.files.is_empty() {
+    if args.workspace || args.folder {
+        // can use the same code for both
+        let res = || -> Result<()> {
+            {
+                let inst = time::Instant::now();
+                eprintln!("\x1B[1;32m  Formatting\x1B[0;22m {}...", if args.folder {
+                    "folder"
+                } else {
+                    "workspace"
+                });
+                let mut workspace_cargo_toml = None;
+                let c_dir = current_dir()?;
+                if args.workspace {
+                    let mut at: Option<&Path> = Some(&c_dir);
+                    while let Some(d) = at.take() {
+                        if d.join("Cargo.toml").exists() &&
+                            cargo_manifest::Manifest::from_path(d.join("Cargo.toml"))
+                                .ok()
+                                .and_then(|m| m.workspace)
```

Comment:
Since this ends up processing the Cargo.toml even if it doesn't have a workspace below, I think it should be fine to skip this check.

If this skipped the actual Cargo.toml and ended up formatting a parent project's files too that would be pretty unexpected I think.

Reply:
Ah, I see, that's only if the folder option is used... still, I think it would be safer to never ignore a Cargo.toml file.